libsolidity/codegen/ir/IRGenerationContext.cpp

```diff
@@ -121,10 +121,31 @@ string IRGenerationContext::newYulVariable()
 	return "_" + to_string(++m_varCounter);
 }
 
-string IRGenerationContext::internalDispatch(YulArity const& _arity)
+string IRGenerationContext::generateInternalDispatchFunction(YulArity const& _arity)
 {
-	string funName = "dispatch_internal_in_" + to_string(_arity.in) + "_out_" + to_string(_arity.out);
-	return m_functions.createFunction(funName, [&]() {
+	// UNIMPLEMENTED: Internal library calls via pointers are not implemented yet.
+	// We're not generating code for internal library functions here even though it's possible
+	// to call them via pointers. Right now such calls end up triggering the `default` case in
+	// the switch above.
+	set<FunctionDefinition const*> functions;
+	for (auto const& contract: mostDerivedContract().annotation().linearizedBaseContracts)
+		for (FunctionDefinition const* function: contract->definedFunctions())
+			if (
+				!function->isConstructor() &&
+				YulArity::fromType(*TypeProvider::function(*function, FunctionType::Kind::Internal)) == _arity
+			)
+			{
+				functions.insert(function);
+				enqueueFunctionForCodeGeneration(*function);
+			}
+
+	return generateInternalDispatchFunction(_arity, move(functions));
+}
+
+string IRGenerationContext::generateInternalDispatchFunction(YulArity const& _arity, set<FunctionDefinition const*> _functions)
+{
+	string funName = IRNames::internalDispatch(_arity);
+	return m_functions.createFunction(funName, [funName, _arity, functions(move(_functions))]() {
```

Comment:
Why again can't we capture `_functions` by const ref?

Reply:
OK, looks like we can :) I wanted to create a counterexample and then remembered that `createFunction()` executes the lambda immediately instead of storing it for later. So having references to local variables is fine since they're still on the stack.

I removed the `move`.